### DISCUSSION TITLE
How to run GitHub Action as non-root user?

### DISCUSSION THREAD

**Discussion by stephen-crawford** [community/community#37148] [Actions]
Hello everyone, 

I am trying to run a GitHub Action using Powershell commands for CI/CD. As part of the requirements for the project, we need to be able to run the application without being a root user (we want to avoid potential RCE problems). That being said, I cannot seem to find how to modify either the Powershell commands or workflow details to be able to run the .bat file as a non-root user during the actions workflow.

Powershell has the ability to create users, but it is not clear how to do the equivalent:
`RUN useradd -u 1001 -s /sbin/nologin me`
`ENV PATH="/me/bin:${PATH}` 
in a manner that the GitHub Action can then execute the program as the new user (me). 

Does anyone know how//if this is possible? 

Thank you.

Tagging @cwperks so he is also aware of any help.

**Comment by ossanna16** [comment]
Hi there @scrawfor99 and welcome to our community! Thank you for asking a great question 🙂

**Reply by stephen-crawford** [reply]
Thank you for having me. For the time being, we ended up reverting our change that required running in admin. @cwperks and I will keep everyone updated should we end up figuring out a solution that allows this functionality in full.

**Comment by cwperks** [comment]
This is an issue we are facing on the `windows-latest` runner where all commands are run as admin. 

It seems like the correct approach to use would be:

```
net user "username" "<password>" /add
runas /user:username ".\bin\program.bat"
```

But `runas` is designed to prompt for password input and there is no way to supply a password. passwordless is also unsupported.
